src/box/lua/config/utils/aboard.lua

```diff
@@ -0,0 +1,178 @@
+-- modifies it.
+--
+-- A timestamp is saved on the :set() method call to show it in
+-- an :alerts() result.
+local function aboard_set(self, alert, opts)
+    assert(alert.type == 'error' or alert.type == 'warn')
+    if alert.type == 'error' then
+        log.error(alert.message)
+    else
+        log.warn(alert.message)
+    end
+    alert.timestamp = datetime.now()
+
+    local key
+    if opts == nil or opts.key == nil then
+        key = self._next_key
+        self._next_key = self._next_key + 1
+    else
+        key = opts.key
+    end
+
+    self._alerts[key] = alert
+end
+
+-- Return an alert pointed by the given `key` or nil.
+local function aboard_get(self, key)
+    return self._alerts[key]
+end
+
+-- Drop an alert pointed by the given `key`.
+--
+-- The function is no-op if the alert doesn't exist.
+--
+-- The `on_drop` callback (see the aboard.new() function) is
+-- called if the alert had exist.
+local function aboard_drop(self, key)
+    if self._alerts[key] == nil then
+        return
+    end
+    self._alerts[key] = nil
+    if self._on_drop ~= nil then
+        self._on_drop(self, key)
+    end
+end
+
+-- Drop all the alerts.
+--
+-- The `on_drop` callback is NOT called.
+local function aboard_clean(self)
+    self._alerts = {}
+    self._next_key = 1
+end
+
+-- Serialize the alerts to show them to a user.
+--
+-- The alerts are sorted by a time of the :set() method call and
+-- returned as an array-like table.
+--
+-- The keys passed to the :set() method are NOT shown in the
+-- result of this method: they're considered as internal
+-- identifiers.
+--
+-- Fields of the alert object that start from the underscore
+-- character are NOT shown.
+local function aboard_alerts(self)
+    -- Don't return alert keys.
+    local alerts = {}
+    for _, alert in pairs(self._alerts) do
+        -- Don't show fields that start from an underscore.
+        local serialized = {}
+        for k, v in pairs(alert) do
+            if not k:startswith('_') then
+                serialized[k] = v
+            end
+        end
+        table.insert(alerts, serialized)
+    end
+    -- Sort by timestamps.
+    table.sort(alerts, function(a, b)
+        return a.timestamp < b.timestamp
+    end)
+    return alerts
+end
+
+-- Return one of three possible statuses:
+--
+-- * ready
+-- * check_warnings
+-- * check_errors
+local function aboard_status(self)
+    local status = 'ready'
+
+    for _, alert in pairs(self._alerts) do
+        assert(alert.type == 'error' or alert.type == 'warn')
+        if alert.type == 'error' then
+            return 'check_errors'
+        end
+
+        status = 'check_warnings'
+    end
+
+    return status
+end
+
+-- Return `true` if there are no alerts, `false` otherwise.
+local function aboard_is_empty(self)
+    return next(self._alerts) == nil
+end
+
+-- Traverse over alerts with their internal identifiers (keys).
+local function aboard_pairs(self)
+    return pairs(self._alerts)
+end
```

Comment:
`aboard_pairs()` is introduced, but never used in the patchset.

Reply:
It is called as `config._aboard:pairs()` in [sync_alerts](https://github.com/tarantool/tarantool/commit/6ae9a0a55419ef2f95705a24a0064d201cb836a1#diff-b71368783dfc0d7f6c89c69d145494c61e9e5432f97e6895beadf1767da139a7R612).

Reply:
Oh, missed that, thank you.